java/demo/src/com/google/phonenumbers/PhoneNumberParserServlet.java

```diff
@@ -322,6 +324,7 @@ private StringBuilder getOutputForSingleNumber(
     } catch (NumberParseException e) {
       output.append(e.toString());
```

Comment:
This is wrong because the error string can appear in the middle of the output (and may also contain inadvertent metacharacters). Change this so that the NumberParseException can only happen _before_ we do any output writing. Then remove any try/catch from the output code (so any errors just kill the request).